src/logic.ts

```diff
@@ -0,0 +1,13 @@
+// This module of solid-ui has a main quadstore for the app to use
+//
+
+import * as debug from './debug'
+import { fetch } from 'solid-auth-client'
+// import { SolidLogic } from 'solid-logic'
+import { SolidLogic } from './solid-logic-move-me'
+
+export const solidLogicSingleton = new SolidLogic({ fetch })
```

Comment:
I dislike the idea of having a global god singleton object containing all the logic. But if it is only a migration step into the idea of separating logic from ui I guess it is a valid start

Reply:
Well, it's a singleton instantiation of another code repo. In other code (e.g. Solid test-suite), I would probably instantiate multiple of these, but for the mashlib philosophy we want all data to go into one big searchable stack.
You are aware that the current code has a global god singleton object for the store, in `src/store.ts`, right? That's basically what this sits on on top of.

Reply:
I am fine with a single store as single source of truth. This does not mean, all logic has to be in one object or class. There are several concerns that should be split in the long run.

Reply:
Right, I see your point. So how about something like a factory pattern, where you have a single SolidLogic object around the store, but you use it to instantiate various objects of different classes (e.g. CurrentUser, AddressBookEntry, etc) that get an internal reference to the store when you instantiate them?

Reply:
sounds good

Reply:
It's fine for me if you want to merge first and do it in a separate PR